### PR TITLE
add an LDAP host label

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Adds a Label on synced groups containing the LDAP host IP. Reconfigures OpenShift lister to use this instead of getting all groups and filtering on annotation.

cc/ @deads2k

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@deads2k Added annotation filter.

**Comment by stevekuznetsov** [comment]
[test][extended:authentication]

File: pkg/cmd/experimental/syncgroups/grouplister_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	listFailClient := testclient.NewSimpleFake()
+	listFailClient.PrependReactor("list", "groups", func(action ktestclient.Action) (bool, runtime.Object, error) {
+		return false, nil, errors.New("failed to list groups")
+	})
+
+	lister := NewAllOpenShiftGroupLister([]string{}, "test-host:port", listFailClient.Groups())
+	groupUIDs, err := lister.ListGroups()
+	if err == nil {
+		t.Error("expected an error listing groups, got none")
+	} else if expected, actual := "failed to list groups", err.Error(); expected != actual {
```
Comment (deads2k) [inline pkg/cmd/experimental/syncgroups/grouplister_test.go]: if you're going to do this, `e` and `a` are good.

File: pkg/cmd/experimental/syncgroups/grouplister_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		if expected, actual := []string{testCase.expectedName}, groupNames; err == nil && !reflect.DeepEqual(expected, actual) {
+			t.Errorf("%s: expected UIDs %v, got %v", name, expected, actual)
+		}
+	}
+}
+
+func TestListAllOpenShiftGroupsListErr(t *testing.T) {
+	listFailClient := testclient.NewSimpleFake()
+	listFailClient.PrependReactor("list", "groups", func(action ktestclient.Action) (bool, runtime.Object, error) {
+		return false, nil, errors.New("failed to list groups")
```
Comment (deads2k) [inline pkg/cmd/experimental/syncgroups/grouplister_test.go]: `bool` is for "handled".  return true

**Comment by stevekuznetsov** [comment]
subsumed into https://github.com/openshift/origin/pull/5108

**Pull request closed without merging**